MonoGame.Framework/Graphics/Texture2D.OpenGL.cs

```diff
@@ -495,13 +479,12 @@ private void PlatformSaveAsPng(Stream stream, int width, int height)
 #if ANDROID
             SaveAsImage(stream, width, height, Bitmap.CompressFormat.Png);
 #else
-            var pngWriter = new PngWriter();
-            pngWriter.Write(this, stream);
+            SaveAsImage(stream, width, height, ImageWriterType.Png);
 #endif
         }
 
 #if DESKTOPGL || MONOMAC
-        internal void SaveAsImage(Stream stream, int width, int height, ImageFormat format)
+        internal void SaveAsImage(Stream stream, int width, int height, ImageWriterType format)
```

Comment:
You modified a piece of code that is used by both DESKTOPGL and MONOMAC, correct thing would be to add `#elif DESKTOPGL` and add code bellow.